source/Weapon.cpp

```diff
@@ -17,21 +96,121 @@ void Weapon::onPose(Array<shared_ptr<Surface> >& surface) {
 	}
 }
 
+void Weapon::simulateProjectiles(SimTime sdt, const Array<shared_ptr<TargetEntity>>& targets, const Array<shared_ptr<Entity>>& dontHit) {
+	// Iterate through projectiles for hit/miss detection here
+	for (int p = 0; p < m_projectiles.size(); p++) {
+		shared_ptr<Projectile> projectile = m_projectiles[p];
+		projectile->onSimulation(sdt);
+		// Remove the projectile for timeout
+		if (projectile->remainingTime() <= 0) {
+			// Expire
+			m_scene->removeEntity(projectile->name());
+			m_projectiles.remove(p);
+			--p;
+		}
+		else if (!m_config->hitScan) {
+			// Distance at which to delcare a hit
+			const float hitThreshold = m_config->bulletSpeed * 2.0f * (float)sdt;
+			// Look for collision with the targets
+			const Ray ray = projectile->getCollisionRay();
+			float closest = finf();
+			Model::HitInfo info;
+			shared_ptr<TargetEntity> closestTarget;
+			for (shared_ptr<TargetEntity> t : targets) {
+				if (t->intersect(ray, closest, info)) {
+					closestTarget = t;
+				}
+			}
+			// Check for target hit
+			if (closest < hitThreshold) {
+				m_hitCallback(closestTarget);
+				// Offset position slightly along normal to avoid Z-fighting the target
+				drawDecal(info.point + 0.01 * info.normal, m_camera->frame().lookVector(), true);
+				projectile->clearRemainingTime();
+			}
+			// Handle (miss) decals here
+			else {
+				Array<shared_ptr<Entity>> dontHitItems = dontHit;
+				dontHitItems.append(m_currentMissDecals);
+				dontHitItems.append(targets);
+				dontHitItems.append(m_projectiles);
+				// Check for closest hit (in scene, otherwise this ray hits the skybox)
+				//closest = finf();
+				const Ray ray = projectile->getDecalRay();
+				m_scene->intersect(ray, closest, false, dontHitItems, info);
+
+				// If we are within 2 simulation cycles of a wall, create the decal
+				if (closest < hitThreshold) {
+					// Offset position slightly along normal to avoid Z-fighting the wall
+					drawDecal(info.point + 0.01 * info.normal, info.normal);
+					projectile->clearRemainingTime();							// Stop the projectile here
+					m_missCallback();
+					//sess->accumulatePlayerAction(PlayerActionType::Miss);		// Declare this shot a miss here
```

Comment:
Should this commented out line stay here?

Reply:
No, more of a reminder to thoroughly test player action logs. Can be removed.